### PR TITLE
[AD-267] Add SQL and JDBC support and limitations documentation

### PR DESCRIPTION
**Pull request by andiemontoyeah**
### Summary

- Add SQL and JDBC support and limitations documentation

### Description

- Separated SQL and JDBC support into 2 pages 
- For SQL: Added supported operators, supported data types, supported casts, identifier limitations 
- For JDBC: Added list of unsupported methods for various interfaces

### Related Issue

https://bitquill.atlassian.net/browse/AD-302

### Additional Reviewers
@birschick-bq
@mitchell-elholm
@alexey-temnikov 
@andiem-bq
@affonsoBQ

### CODE REVIEW

**Review by affonsov — APPROVED** [review]

**Review by alexey-temnikov — APPROVED** [review]

File: src/markdown/jdbc/jdbc-limitations.md [hunk 1]
```diff
@@ -0,0 +1,173 @@
+# JDBC Support and Limitations
+
+## Schema, Catalog, and Tables
+The semantics of schema, catalogs, and tables often vary by database vendor.
+For the DocumentDB JDBC driver, the "schema" is a particular database on a DocumentDB cluster, and
+the "tables" are determined from the collections within that database 
+through [schema discovery or generation](/src/markdown/schema/schema-discovery.md).
```
Comment (birschick-bq) [inline src/markdown/jdbc/jdbc-limitations.md]: schema discovery **and** generation?

File: src/markdown/sql/sql-limitations.md [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+some common extensions. 
+This section highlights limitations to keep in mind when constructing SQL queries for the driver.
+
+## Basic Query Format
+The driver supports `SELECT` statements of the general form: 
+```
+SELECT [ ALL | DISTINCT ] { * | projectItem [, projectItem ]* }
+   FROM tableExpression
+   [ WHERE booleanExpression ]
+   [ GROUP BY { groupItem [, groupItem ]* } ]
```
Comment (birschick-bq) [inline src/markdown/sql/sql-limitations.md:12]: I don't understand the syntax with curly braces (`{` and `}`). It is not used elsewhere in the syntax.

File: src/markdown/sql/sql-limitations.md [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+- `VARCHAR`
+- `BINARY`
+- `VARBINARY`
+- `DATE`
+- `TIME`
+- `TIMESTAMP`
+
+Note that while all the above can be used when constructing queries, columns themselves can 
+only be of the types `BOOLEAN`, `BIGINT`, `INTEGER`, `DECIMAL`, `DOUBLE`, `VARCHAR`, `VARBINARY` and `TIMESTAMP`. 
+See the [schema discovery or generation](/src/markdown/schema/schema-discovery.md) sections for more information.
```
Comment (birschick-bq) [inline src/markdown/sql/sql-limitations.md]: schema discovery **and** generation?

File: src/markdown/sql/sql-limitations.md [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+| CHAR, VARCHAR                   | N       | N                              | N                            | N    | N    | N         | _             | N
+| BINARY, VARBINARY               | N       | N                              | N                            | N    | N    | N         | N             | _
+
+The driver also allows for implicit conversions when such a conversion makes sense. For example, a `DATE` value passed to 
+a function that takes `TIMESTAMP`, will be automatically cast to `TIMESTAMP`.
+
+## Operators and Functions 
+
+### Comparison Operators
+- `value1  < op > value2`  where `op` is one of : `=`, `<>`, `<`, `>`, `<=` or `>=`
```
Comment (birschick-bq) [inline src/markdown/sql/sql-limitations.md]: Can you remove the space between `<` and `op` and between `op` and `>`?

File: src/markdown/sql/sql-limitations.md [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+Functions that accept a `timestamp` value that are passed a `DATE` type will implicitly convert the value to 
+a `TIMESTAMP`, padding the time portion with 0s.
+
+`timeUnit` can be one of `YEAR` | `QUARTER` | `MONTH` | `WEEK` | `DOY` | `DOW` | `DAY` | `HOUR` | `MINUTE` 
+| `SECOND` | `MICROSECOND` .
+
+Note that `EXTRACT` for `WEEK` or `WEEK()` return an integer between 0 and 53. This is in line with the MongoDB
+interpretation of week numbers but other databases more commonly return a value between 1 and 53.
+
+Addition using the `YEAR`, `QUARTER`, and `MONTH` intervals is not supported in `TIMESTAMPADD`.
```
Comment (birschick-bq) [inline src/markdown/sql/sql-limitations.md]: intervals **are** not supported

File: src/markdown/sql/sql-limitations.md (hunk 2 above)
Comment (andiemontoyeah) [reply src/markdown/sql/sql-limitations.md:12]: hmm i guess i was using [ ] to denote a part that is optional and i wanted { } to be a part that is not optional but has a flexible format: { * | projectItem [, projectItem ]* } . Should i just remove the { }  enclosing these parts?

Comment (birschick-bq) [reply src/markdown/sql/sql-limitations.md:12]: Sure. Understand. Keep it as is. Thanks for the explanation.

**Review by birschick-bq — APPROVED** [review]

**Pull request merged**